form-flow/src/main/kotlin/com/ritense/formflow/service/ObjectMapperConfigurer.kt

```diff
@@ -18,26 +18,25 @@ package com.ritense.formflow.service
 
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.fasterxml.jackson.databind.jsontype.NamedType
-import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
 import com.fasterxml.jackson.module.kotlin.registerKotlinModule
 import com.ritense.formflow.domain.definition.configuration.step.StepTypeProperties
+import org.springframework.boot.context.event.ApplicationStartedEvent
+import org.springframework.context.event.EventListener
 
-class FormFlowObjectMapper(
-    objectMapper:ObjectMapper = jacksonObjectMapper(),
-    stepPropertiesTypes: Collection<NamedType>? = null
+class ObjectMapperConfigurer(
+    private val objectMapper: ObjectMapper,
+    private val stepPropertiesTypes: Collection<NamedType>
 ) {
 
-    private val mapper = objectMapper.copy()
-        .registerKotlinModule()
-        .apply {
-            stepPropertiesTypes
-                ?.filter { StepTypeProperties::class.java.isAssignableFrom(it.type) }
-                ?.forEach {
-                    registerSubtypes(it)
-                }
-        }
-
-    fun get(): ObjectMapper {
-        return mapper
+    @EventListener(ApplicationStartedEvent::class)
+    fun configure() {
+        objectMapper.registerKotlinModule()
```

Comment:
You're actually changing the application-wide objectmapper like this, as the `registerKotlinModule` is mutating `this`. Not sure if that's the desired effect.

Reply:
That is what i intended to do. Maybe the kotlin module itself doesn't have to be in this configuration.